restore.py

```diff
@@ -209,8 +214,7 @@ def restore_partitions():
             # restore boot loader
             if boot_config.src_fmt == 'grub2':
                 if efi_boot:
-                    branding = util.readKeyValueFile(os.path.join(backup_fs.mount_point, constants.INVENTORY_FILE))
-                    branding['product-brand'] = branding['PRODUCT_BRAND']
+                    mounts['esp'] = esp
```

Comment:
As you pointed out on https://github.com/xenserver/host-installer/pull/161/files, these solve the same issue
I think adding `esp` to `mounts` should happen when the mounting occurs in case any future changes need to inspect what is mounted and rely on that `mounts` dictionary
With your change here, it would incorrectly appear as if `esp` is not mounted until this code is run
(Another improvement would be to use `'esp' in mounts` instead of `if efi_mounted` but that's for another PR)